executor/runtime/types/sidecars.go

```diff
@@ -153,6 +163,18 @@ func shouldStartTitusSeccompAgent(cfg *config.Config, c Container) bool {
 	return c.SeccompAgentEnabledForPerfSyscalls() || c.EffectiveNetworkMode() == titus.NetworkConfiguration_Ipv6AndIpv4Fallback.String() || c.TrafficSteeringEnabled()
 }
 
+func shouldStartSystemDNS(cfg *config.Config, c Container) bool {
+	enabled := cfg.ContainerSystemDNS
+	if !enabled {
+		return false
+	}
+
+	if cfg.SystemDNSServiceImage == "" {
+		return false
+	}
```

Comment:
Hmmmmm....

This needs to only be enabled on the v6 mode.

I don't know if that is available in the `Container` object. If not, you might have to plumb it through.